libc/string/string_cmp.c

```diff
@@ -0,0 +1,439 @@
+	TEST_ASSERT_NOT_EQUAL_INT(0, memcmp(empty, separated, sizeof(separated)));
+
+	free(asciiSet);
+}
+
+
+TEST(string_memcmp, big)
+{
+	char *hugeStr = testdata_createCharStr(BIG_SIZE),
+		 hugeStr2[BIG_SIZE];
+
+	memcpy(hugeStr2, hugeStr, BIG_SIZE);
+
+	TEST_ASSERT_EQUAL_INT(0, memcmp(hugeStr, hugeStr, BIG_SIZE));
+	/* Comparing the same strings, that are placed in different location */
+	TEST_ASSERT_EQUAL_INT(0, memcmp(hugeStr, hugeStr2, BIG_SIZE));
+	hugeStr[BIG_SIZE - 2] = 1;
+	hugeStr2[sizeof(hugeStr2) - 2] = 2;
+	TEST_ASSERT_LESS_THAN_INT(0, memcmp(hugeStr, hugeStr2, BIG_SIZE));
+	TEST_ASSERT_GREATER_THAN_INT(0, memcmp(hugeStr2, hugeStr, BIG_SIZE));
+
+	free(hugeStr);
+}
+
+
+TEST(string_memcmp, various_sizes)
+{
+	int i;
+	/* +1 to provide all ascii characters (ChrStr contains double 1 at the beginning) */
+	char *asciiStr = testdata_createCharStr(BUFF_SIZE + 1);  // not sure whether const is possible there
+	char asciiStr2[BUFF_SIZE + 1];
+
+	memcpy(asciiStr2, asciiStr, sizeof(asciiStr2));
+	for (i = 1; i < BUFF_SIZE; i++) {
+		asciiStr2[i] = asciiStr[i] - 1;
+		TEST_ASSERT_EQUAL_INT(0, memcmp(asciiStr, asciiStr2, i));
+		TEST_ASSERT_GREATER_THAN_INT(0, memcmp(asciiStr, asciiStr2, i + 1));
+		asciiStr2[i] = asciiStr[i];
+	}
+
+	free(asciiStr);
+}
+
+
+TEST(string_memcmp, offsets)
+{
+
+	char dataSet[4000] = { 0 },
+		 supportSet[4000] = { 0 };
+	int s1Offs, s2Offs, szOffset, sz = sizeof(dataSet);
+
+	for (s1Offs = 0; s1Offs < sz; s1Offs++) {
+		dataSet[s1Offs] = s1Offs;
+		supportSet[s1Offs] = s1Offs;
+	}
+
+	/*Testing different offset of data blocks with same space or different*/
+	for (s1Offs = 0; s1Offs < 8; s1Offs++) {
+		for (s2Offs = 0; s2Offs < 8; s2Offs++) {
+			for (szOffset = 0; szOffset < 8; szOffset++) {
+
+				if (s2Offs < s1Offs) {
+					TEST_ASSERT_GREATER_THAN_INT(0, memcmp(&dataSet[s1Offs], &dataSet[s2Offs], sz - (s1Offs + szOffset)));
+					TEST_ASSERT_GREATER_THAN_INT(0, memcmp(&dataSet[s1Offs], &supportSet[s2Offs], sz - (s1Offs + szOffset)));
+				}
+				else if (s2Offs == s1Offs) {
+					TEST_ASSERT_EQUAL_INT(0, memcmp(&dataSet[s1Offs], &dataSet[s2Offs], sz - (s1Offs + s2Offs + szOffset)));
+					TEST_ASSERT_EQUAL_INT(0, memcmp(&dataSet[s1Offs], &supportSet[s2Offs], sz - (s1Offs + s2Offs + szOffset)));
+				}
+				else {
+					TEST_ASSERT_LESS_THAN_INT(0, memcmp(&dataSet[s1Offs], &dataSet[s2Offs], sz - (s2Offs + szOffset)));
+					TEST_ASSERT_LESS_THAN_INT(0, memcmp(&dataSet[s1Offs], &supportSet[s2Offs], sz - (s2Offs + szOffset)));
+				}
+			}
+		}
+	}
+}
+
+
+/*
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+*/
+
+
+TEST_SETUP(string_strncmp)
+{
+}
+
+
+TEST_TEAR_DOWN(string_strncmp)
+{
+}
+
+
+TEST(string_strncmp, basic)
+{
+	const char *basicStr = "Basic";
+
+	TEST_ASSERT_EQUAL_INT(0, strncmp(basicStr, basicStr, 6));
+	TEST_ASSERT_GREATER_THAN_INT(0, strncmp(basicStr, support, 6));
+	TEST_ASSERT_LESS_THAN_INT(0, strncmp(support, basicStr, 6));
```

Comment:
I think you can also use "Test" or sth like that and we could remove `support` buff.